### PR TITLE
chore: add network integration test with kademilia instead mDNS

### PR DESCRIPTION
**Pull request by P1sar**
## Changes
While I was debugging network module I found out that our network tests using mDNS instead of Kademlia and so bypass or uses logic that is not actually used in Kademilia so I decided to implement integration test that covers producion code. 

## Tests
`MODE=network go test ./tests/network/netwrok_integration_test.go -v`

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Issues
Related: https://github.com/ChainSafe/gossamer/issues/3758

## Primary Reviewer

@

### CODE REVIEW

File: dot/network/discovery.go [hunk 1]
```diff
@@ -233,6 +223,13 @@ func (d *discovery) findPeers() {
 	}
 }
 
-func (d *discovery) findPeer(peerID peer.ID) (peer.AddrInfo, error) {
-	return d.dht.FindPeer(d.ctx, peerID)
+func (d *discovery) stop() error {
```
Comment (jimjbrettj) [inline dot/network/discovery.go:226]: was there a reason to move this function?

File: dot/network/service.go [hunk 2]
```diff
@@ -292,14 +292,11 @@ func (s *Service) Start() error {
...
 	// connectHandler is a part of libp2p.Notifiee interface implementation and getting called in the very end
-	//after or Incoming or Outgoing node is connected
+	// after or Incoming or Outgoing node is connected.
 	s.host.cm.connectHandler = func(peerID peer.ID) {
 		for _, prtl := range s.notificationsProtocols {
 			prtl.peersData.setMutex(peerID)
 		}
-		// TODO: currently we only have one set so setID is 0, change this once we have more set in peerSet
-		const setID = 0
-		s.host.cm.peerSetHandler.Incoming(setID, peerID)
```
Comment (jimjbrettj) [inline dot/network/service.go:302]: We do not need this code anymore?

**Comment by jimjbrettj** [comment]
nit/rename file to ./tests/network/network_integration_test.go

File: tests/network/netwrok_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	"github.com/adrg/xdg"
+	"github.com/stretchr/testify/require"
+
+	"context"
+	"os"
+	"testing"
+	"time"
+)
+
+func TestKadDHTNetworkDiscovery(t *testing.T) {
```
Comment (jimjbrettj) [inline tests/network/netwrok_integration_test.go]: This test actually does not work locally for me and it times out. Based on my logs it seems that Alice progresses in the network but Bob and Charlie do not

File: dot/network/discovery.go (hunk 1 above)
Comment (P1sar) [reply dot/network/discovery.go:226]: I have removed discovery) findPeer justt to make  service smaller, there is no reason to keep this as a function when we can call it directly. 
I moved stop to the very end to give logic to method sort, I usually like to have stop in the very end and start in the begging of the file. But I guess it is pretty subjective

File: tests/network/netwrok_integration_test.go (hunk 3 above)
Comment (P1sar) [reply tests/network/netwrok_integration_test.go]: hmmm. how do you run it? it works fine for me, trying to undersrtand what I missed

File: dot/network/service.go [hunk 4]
```diff
@@ -711,7 +708,7 @@ func (s *Service) processMessage(msg peerset.Message) {
 		addrInfo := s.host.p2pHost.Peerstore().PeerInfo(peerID)
 		if len(addrInfo.Addrs) == 0 {
 			var err error
-			addrInfo, err = s.host.discovery.findPeer(peerID)
+			addrInfo, err = s.host.discovery.dht.FindPeer(s.host.discovery.ctx, peerID)
```
Comment (EclesioMeloJunior) [inline dot/network/service.go]: I would suggest to create a new context with timeout here then we have a limit while finding a peer
```suggestion
                         ctx, cancel := context.WithTimeout(s.host.discovery.ctx, _some_timeout_limit_)
                         defer cancel()
			addrInfo, err = s.host.discovery.dht.FindPeer(ctx, peerID)
```

File: tests/network/netwrok_integration_test.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+	con.Core.Role = common.FullNodeRole
+	con.RPC.Modules = []string{"system", "author", "chain"}
+	con.Network.MinPeers = 1
+	con.Network.MaxPeers = 20
+	con.Network.NoMDNS = true // Turning off mDNS, purpose of this test is purly use only kadDHT discovery
+	con.Core.BabeAuthority = true
+	con.Log.Sync = "trace"
+	con.Network.Port = 7001
+	// ID: 12D3KooWMHixgmjFYM4VyQNDTKMvN9BPw47Tyyb6LPZ43EavV68m
+	con.BasePath = xdg.DataHome + "/gossamer/westend-local/alice"
```
Comment (EclesioMeloJunior) [inline tests/network/netwrok_integration_test.go]: I would suggest to use `t.TempDir()` to create an ephemeral dir where we can test

**Review by EclesioMeloJunior — CHANGES_REQUESTED** [review]

File: dot/network/service.go (hunk 2 above)
Comment (P1sar) [reply dot/network/service.go:302]: I was testing this part local and had some doubts about necessety of this part although eventually decided to return it back and rather remove it in separate PR with additional tests.

**Review by EclesioMeloJunior — APPROVED** [review]
LGTM

File: dot/network/discovery.go (hunk 1 above)
Comment (jimjbrettj) [reply dot/network/discovery.go:226]: I like it! I am good with this change was just curious

File: tests/network/netwrok_integration_test.go (hunk 3 above)
Comment (jimjbrettj) [reply tests/network/netwrok_integration_test.go]: retried and seems to be good now

**Review by jimjbrettj — APPROVED** [review]
lgtm

**Pull request merged**